api/src/main/resources/liquibase-schema-only.xml

```diff
@@ -2429,6 +2429,12 @@
     <changeSet author="ben (generated)" id="1227303685425-75">
         <addPrimaryKey columnNames="concept_id, concept_set" tableName="concept_set_derived"/>
     </changeSet>
+    <changeSet author="vshankar" id="201609171146-1">
```

Comment:
This file only gets executed when installing OpenMRS and never again, so any changesets you add to never get executed for installations performing an upgrade yet it's actually meant for them, this is why it's appearing to you like it works when you move it here when actually what's happening is that it's not getting run.

The issue might be because you're providing a blank constraint name, you might want to just get rid of the constraintName attribute